internal/kldevents/eventstream.go

```diff
@@ -272,22 +354,34 @@ func (a *eventStream) eventPoller() {
 				}
 			}
 		}
-		time.Sleep(a.pollingInterval)
+		// the event poller reacts to notification about a stream update, else it starts
+		// another round of polling after completion of the pollingInterval
+		select {
+		case <-a.updateInterrupt:
```

Comment:
Without an error catch here, won't this panic when the stream is closed?

Reply:
Or just make it a `bool` stream, and push a boolean into it?

Reply:
I guess there might be multiple consumers on it, so `close` is an option that allows the same channel to be used in multiple places

Reply:
Discussed on the phone. We've confirmed in a `select` you don't need to explicitly add the second parameter, in order to get triggered with `nil`. If this did a panic, then the tests would be failing.